File: 0788-rotated-digits/0788-rotated-digits.cpp
class Solution {
public:
    int rotatedDigits(int n) {
        int ans = 0;
        for(int i = 1; i <= n; ++i) 
            ans += helper(i);
        return ans;
    }
    
    bool helper(int x) {
        int flag = 0;
        while(x) {
            int t = x % 10;
            x /= 10;
            if(t == 4 or t == 7 or t == 3) return false;
            if(t == 2 or t == 5 or t == 6 or t == 9) flag = 1;
        }
        return flag;
    }
};

File: 0788-rotated-digits/0788-rotated-digits.js
/**
 * @param {number} n
 * @return {number}
 */
var rotatedDigits = function(n) {
    let ans = 0;
    for(let i = 0; i <= n; ++i) {
        if(helper(i)) ++ans;
    }
    return ans;
}

function helper(x) {
    let flag = false;
    while(x > 0) {
        let t = x % 10;
        x = Math.floor(x / 10);
        if(t == 4 || t == 3 || t == 7) return false;
        if(t == 6 || t == 9 || t == 2 || t == 5) flag = true;
    }
    return flag;
}

